.github/workflows/main.yaml

```diff
@@ -58,6 +58,20 @@ jobs:
     - name: Test Python
       run: bash ./pythia test
 
+    - name: Setup SSH key and known_hosts
+      env:
+        SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+      run: |
+        mkdir -p ~/.ssh
+        ssh-keyscan -p 13022 binary.chevah.com >> ~/.ssh/known_hosts
+        ssh-agent -a $SSH_AUTH_SOCK
+        ssh-add - <<< "${{ secrets.BINARY_BIN_KEY }}"
+
+    - name: Upload testing package
+      env:
+        SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+      run: rsync -avhP -e 'ssh -p 13022' dist/ binary-site@binary.chevah.com:binary.chevah.com/testing/
```

Comment:
I would prefer to push using SFTP or SCP... so that in the future we can replace the OpenSSH server  with SFTPPlus.

At the same time, with using SFTPPlus we will no disclose the port number of our main SSH server.

Reply:
> I would prefer to push using SFTP or SCP... so that in the future we can replace the OpenSSH server with SFTPPlus.

I think we could use SFTPPlus right away if you are fine with having a separate hierarchy for Python 3 packages. Could be possible even without it, but sharing permissions with both `binary_site` and `www-data` would get messy.

> 
> At the same time, with using SFTPPlus we will no disclose the port number of our main SSH server.

But is this a real problem? :-] If yes, we could easily fix it with a repo secret…

Reply:
In the end, I've used a new hierarchy for Python 3 packages, hosted with SFTPPlus.

Reply:
I will let you continue working on this... I see that sftp batch fails... but the workflow step is not failing. 

You can also look at using putty batch mode. We used to uses to for SFTPPlus v1.5 as it was working better then OpenSSH SFTP batch mode.... and AFAIK by defafult putty batch fails by default on errors.

Reply:
There are some commands in the sftp batch that I want to be allowed to fail: the creation of $OS and $ARCH sub-dirs for already existing ones. This is by design, so that new $OS/$ARCH hierarchies are created when uploading the first package for such a combo, e.g `ubuntu2204/x64`.

Or are you referring to something else?

Reply:
You can tell which commands are allowed to fail because they are prefixed with `-`.